tests/ttnn/unit_tests/operations/test_moreh_bmm.py

```diff
@@ -8,18 +8,71 @@
 from loguru import logger
 
 import ttnn
-from tests.ttnn.unit_tests.operations.test_moreh_matmul import get_tensors
 from models.utility_functions import comp_allclose_and_pcc
 
 from tests.ttnn.unit_tests.operations.test_utils import (
     get_compute_kernel_options,
     compute_kernel_options,
     compute_kernel_ids,
-    to_ttnn,
 )
 
 
-def run_moreh_bmm(shape, optional_output, compute_kernel_options, device):
+def create_ttnn_tilized_tensor(torch_tensor, device, dtype):
+    return ttnn.from_torch(torch_tensor, device=device, dtype=dtype, layout=ttnn.TILE_LAYOUT)
+
+
+def get_tensors(
+    input_shape,
+    mat2_shape,
+    output_shape,
+    require_input_grad,
+    require_mat2_grad,
+    device,
+    *,
+    torch_dtype=torch.float32,
+    ttnn_dtype=ttnn.bfloat16,
+):
+    torch_input = torch.rand(input_shape, dtype=torch_dtype)
+    torch_mat2 = torch.rand(mat2_shape, dtype=torch_dtype)
+    torch_output = torch.rand(output_shape, dtype=torch_dtype)
+
+    ttnn_input = create_ttnn_tilized_tensor(torch_input, device, ttnn_dtype)
+    ttnn_mat2 = create_ttnn_tilized_tensor(torch_mat2, device, ttnn_dtype)
+    ttnn_output = create_ttnn_tilized_tensor(torch_output, device, ttnn_dtype)
+
+    ttnn_output_grad = torch_output_grad = ttnn_input_grad = ttnn_mat2_grad = None
+    if require_input_grad or require_mat2_grad:
+        torch_output_grad = torch.rand(output_shape, dtype=torch_dtype)
+        ttnn_output_grad = create_ttnn_tilized_tensor(torch_output_grad, device, ttnn_dtype)
+        if require_input_grad:
+            torch_input_grad = torch.full(input_shape, float("nan"), dtype=torch_dtype)
+            ttnn_input_grad = create_ttnn_tilized_tensor(torch_input_grad, device, ttnn_dtype)
+        if require_mat2_grad:
+            torch_mat2_grad = torch.full(mat2_shape, float("nan"), dtype=torch_dtype)
+            ttnn_mat2_grad = create_ttnn_tilized_tensor(torch_mat2_grad, device, ttnn_dtype)
+
+    return (
+        ttnn_input,
+        ttnn_mat2,
+        ttnn_output,
+        ttnn_output_grad,
+        ttnn_input_grad,
+        ttnn_mat2_grad,
+        torch_input,
+        torch_mat2,
+        torch_output_grad,
+    )
```

Comment:
```suggestion
def get_torch_tensors(
    input_shape,
    mat2_shape,
    output_shape,
    require_input_grad,
    require_mat2_grad,
    *,
    dtype=torch.float32,
):
    input = torch.rand(input_shape, dtype=dtype)
    mat2 = torch.rand(mat2_shape, dtype=dtype)
    output = torch.rand(output_shape, dtype=dtype)

    if require_input_grad or require_mat2_grad:
        output_grad = torch.rand(output_shape, dtype=dtype)
        if require_input_grad:
            input_grad = torch.full(input_shape, float("nan"), dtype=dtype)
        if require_mat2_grad:
            mat2_grad = torch.full(mat2_shape, float("nan"), dtype=dtype)

    return (
        input,
        mat2,
        output,
        output_grad,
        input_grad,
        mat2_grad
    )

def create_ttnn_tilized_tensors(torch_tensors, device=device, dtype=dtype):
    tt_tensors = []
    for torch_tensor in torch_tensors:
        tt_tensors.append(ttnn.from_torch(torch_tensor, device=device, dtype=dtype, layout=ttnn.TILE_LAYOUT))
     return tt_tensors

def get_tt_tensors(
    torch_tensors,
    device,
    *,
    ttnn_dtype=ttnn.bfloat16,
):
    # torch_tensors = [input, mat2, output, output_grad, input_grad, mat2_grad]
    return create_ttnn_tilized_tensors(torch_tensors, device=device, dtype=ttnn_dtype)
```

How about splitting `get_tensors` into `get_torch_tensor` and `get_tt_tensor`?

In the code above, words like `torch_`, `ttnn_`, `create_ttnn_tilized_tensor`, and `device` are repeated with high frequency. I believe this repetitive code makes it difficult to understand the actual program's behavior.

Please check only the concept of the code I proposed. I'm not sure if it actually works.

Reply:
If the modifications are complex, it's okay to skip them. Everything else looks good.

Reply:
I agree with you; it's interesting when I modify this get_tensors. Please take a look at: https://github.com/tenstorrent/tt-metal/pull/14089/commits/abc4f3e15dbacf585f5fd685822d3ba0ae2e911e